Sources/X509/Verifier/TrustRootLoading.swift

```diff
@@ -0,0 +1,98 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCertificates open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftCertificates project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftCertificates project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftASN1
+
+struct TrustRootsLoadingError: Error {
+    var errors: [(path: String, error: any Error)]
+}
+
+#if os(Linux)
+/// This is a list of root CA file search paths. This list contains paths as validated against several distributions.
+/// If you are attempting to use swift-certificates on a platform that is not covered here and certificate validation is
+/// failing, please open a pull request that adds the appropriate search path.
+private let rootCAFileSearchPaths = [
+    "/etc/ssl/certs/ca-certificates.crt",  // Ubuntu, Debian, Arch, Alpine,
+    "/etc/pki/tls/certs/ca-bundle.crt",  // Fedora
+]
+
+extension CertificateStore {
+    /// A ``CertificateStore`` that includes all root Certificate Authorities (CAs) that
+    /// are installed in the systems trust store.
+    ///
+    /// You can add additional trust roots by ``CertificateStore/inserting(_:)-5sc2d`` them into the returned ``CertificateStore``.
+    ///
+    /// - Note: Access this property as early as possible. It will start loading and parsing of the certificates in the background.
+    /// Accessing this property does **not** block.
+    /// - Warning: This property is only available on Linux.
```

Comment:
We shouldn't do that: we should make it available everywhere, even if it isn't very useful. Generally we avoid having API only available on one platform. However, we should still have a warning that says that on Apple platforms and on Windows this will not actually find the system trust store, and that the platform verifier should be used instead.